packages/eddsa-poseidon/src/eddsa-poseidon.ts

```diff
@@ -177,23 +189,46 @@ export function unpackPublicKey(publicKey: BigNumber): Point<string> {
     return [unpackedPublicKey[0].toString(), unpackedPublicKey[1].toString()]
 }
 
+/**
+ * Represents a cryptographic entity capable of signing messages and verifying signatures
+ * using the EdDSA scheme with Poseidon hash and the Baby Jubjub elliptic curve.
+ */
 export class EdDSAPoseidon {
+    // Private key for signing, stored securely.
     privateKey: BigNumberish
+    // The secret scalar derived from the private key.
```

Comment:
Worth adding some comments about why this attribute is defined.

Reply:
"The secret scalar derived from the private key __to compute the public key__".

This should be sufficient, as more information can be retrieved using the `deriveSecretScalar()` function. What do you think @cedoor?

Reply:
```ts
/**
 * Derives a secret scalar from a given EdDSA private key.
 * This process involves hashing the private key with Blake1, pruning the resulting hash to retain the lower 32 bytes,
 * and converting it into a little-endian integer. The use of the secret scalar streamlines the public key generation
 * process by omitting steps 1, 2, and 3 as outlined in RFC 8032 section 5.1.5, enhancing circuit efficiency and simplicity.
 * This method is crucial for fixed-base scalar multiplication operations within the correspondent cryptographic circuit.
 * For detailed steps, see: {@link https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.5}.
 * @param privateKey The EdDSA private key for generating the associated public key.
 * @returns The derived secret scalar, optimized for circuit calculations.
 */
 ```
 Do you think is worth linking the circuit as an example of implementation?